### PR TITLE
Refactor Cluster stats code

### PR DESCRIPTION
**Pull request by venkateshamatam**
# Description
Refactor Cluster stats code.

# Testing
- Tested using usage stats from playground
# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/usageStats/stats.go [hunk 1]
```diff
@@ -541,6 +520,77 @@ func GetUsageStats(pastXhours uint64, granularity UsageStatsGranularity, orgid u
...
+	readStats.EventCount, err = strconv.ParseUint(record[1], 10, 64)
+	if err != nil {
+		return readStats, err
+	}
+
+	switch len(record) {
+	case 3:
+		tsString, err := strconv.ParseInt(record[2], 10, 64)
+		if err != nil {
+			return readStats, err
```
Comment (nkunal) [inline pkg/usageStats/stats.go:549]: `fmt.Errorf("parseRecord: could not parse timestamp field: %v", tsString)` could be a better error

File: pkg/usageStats/stats.go [hunk 2]
```diff
@@ -541,6 +520,77 @@ func GetUsageStats(pastXhours uint64, granularity UsageStatsGranularity, orgid u
 	return resultMap, nil
 }
 
+// The data format has evolved over time:
+// - Initially, it was: bytes, eventCount, time
+// - Then, metrics were added: bytes, eventCount, metricCount, time
+// - Later, logsBytesCount and metricsBytesCount were added. However, the new format is backward compatible with the old formats.
+// The current format is: bytes, eventCount, metricCount, time, logsBytesCount, metricsBytesCount
+func parseRecord(record []string) (ReadStats, error) {
```
Comment (nkunal) [inline pkg/usageStats/stats.go:528]: `parseStatsRecord` could be a better name. (dont forget to change it in all the error statements below as well)

File: pkg/usageStats/stats.go [hunk 3]
```diff
@@ -541,6 +520,77 @@ func GetUsageStats(pastXhours uint64, granularity UsageStatsGranularity, orgid u
...
+			return readStats, err
+		}
+		readStats.TimeStamp = time.Unix(tsString, 0)
+		readStats.LogsBytesCount = readStats.BytesCount
+		readStats.MetricsBytesCount = 0
+
+	case 4:
+		readStats.MetricsDatapointsCount, err = strconv.ParseUint(record[2], 10, 64)
+		if err != nil {
+			return readStats, err
```
Comment (nkunal) [inline pkg/usageStats/stats.go:558]: same comment as above

File: pkg/usageStats/stats.go [hunk 4]
```diff
@@ -541,6 +520,77 @@ func GetUsageStats(pastXhours uint64, granularity UsageStatsGranularity, orgid u
...
+		}
+		tsString, err := strconv.ParseInt(record[3], 10, 64)
+		if err != nil {
+			return readStats, err
```
Comment (nkunal) [inline pkg/usageStats/stats.go:562]: same as above

File: pkg/utils/httpserverutils.go [hunk 5]
```diff
@@ -412,18 +412,32 @@ func NewSingleESResponse() *SingleESResponse {
 
 func WriteResponse(ctx *fasthttp.RequestCtx, httpResp HttpServerResponse) {
 	ctx.SetContentType(ContentJson)
-	jval, _ := json.Marshal(httpResp)
-	_, err := ctx.Write(jval)
+	jval, err := json.Marshal(httpResp)
 	if err != nil {
+		log.Errorf("WriteResponse: Error marshaling HTTP response: %v", err)
```
Comment (AndrewHess) [inline pkg/utils/httpserverutils.go:417]: nit: I'm not sure the `err` will include enough info about the `httpResp` to debug what went wrong. If it doesn't, you should include `httpResp` in the log message.

Similar comment for lines 423, 433, 439

**Review by nkunal — APPROVED** [review]

**Pull request merged**